modules/porous_flow/src/materials/PorousFlowFluidStateBaseMaterial.C

```diff
@@ -0,0 +1,197 @@
+        "PorousFlow_mass_frac" + _ext)),
+    _grad_mass_frac_qp(
+        _nodal_material
+            ? nullptr
+            : &this->template declareGenericProperty<std::vector<std::vector<RealGradient>>, is_ad>(
+                  "PorousFlow_grad_mass_frac" + _ext)),
+    _dmass_frac_dvar(
+        is_ad ? nullptr
+              : &this->template declareProperty<std::vector<std::vector<std::vector<Real>>>>(
+                    "dPorousFlow_mass_frac" + _ext + "_dvar")),
+    _fluid_density(this->template declareGenericProperty<std::vector<Real>, is_ad>(
+        "PorousFlow_fluid_phase_density" + _ext)),
+    _dfluid_density_dvar(is_ad ? nullptr
+                               : &this->template declareProperty<std::vector<std::vector<Real>>>(
+                                     "dPorousFlow_fluid_phase_density" + _ext + "_dvar")),
+    _fluid_viscosity(this->template declareGenericProperty<std::vector<Real>, is_ad>(
+        "PorousFlow_viscosity" + _ext)),
+    _dfluid_viscosity_dvar(is_ad ? nullptr
+                                 : &this->template declareProperty<std::vector<std::vector<Real>>>(
+                                       "dPorousFlow_viscosity" + _ext + "_dvar")),
+    _fluid_enthalpy(this->template declareGenericProperty<std::vector<Real>, is_ad>(
+        "PorousFlow_fluid_phase_enthalpy" + _ext)),
+    _dfluid_enthalpy_dvar(is_ad ? nullptr
+                                : &this->template declareProperty<std::vector<std::vector<Real>>>(
+                                      "dPorousFlow_fluid_phase_enthalpy" + _ext + "_dvar")),
+    _fluid_internal_energy(this->template declareGenericProperty<std::vector<Real>, is_ad>(
+        "PorousFlow_fluid_phase_internal_energy" + _ext)),
+    _dfluid_internal_energy_dvar(
+        is_ad ? nullptr
+              : &this->template declareProperty<std::vector<std::vector<Real>>>(
+                    "dPorousFlow_fluid_phase_internal_energy" + _ext + "_dvar"))
+{
+  // Set the size of the FluidStateProperties vector
+  _fsp.resize(_num_phases, FluidStateProperties(_num_components));
+}
+
+template <>
+GenericReal<false>
+PorousFlowFluidStateBaseMaterialTempl<false>::genericValue(const ADReal & value)
+{
+  return MetaPhysicL::raw_value(value);
+}
+
+template <>
+GenericReal<true>
+PorousFlowFluidStateBaseMaterialTempl<true>::genericValue(const ADReal & value)
+{
+  return value;
+}
+
+template <bool is_ad>
+void
+PorousFlowFluidStateBaseMaterialTempl<is_ad>::initQpStatefulProperties()
+{
+  _is_initqp = true;
+  // Set the size of pressure and saturation vectors
+  PorousFlowVariableBaseTempl<is_ad>::initQpStatefulProperties();
+
+  // Set the size of all other vectors
+  setMaterialVectorSize();
+
+  thermophysicalProperties();
+
+  // Set the initial values of the properties at the nodes.
+  // Note: not required for qp materials as no old values at the qps are requested
+  // unless the material is AD (for the FV case there are no nodal materials)
```

Comment:
there is no AD old material properties though, the derivatives wouldnt match the current value of the solution

Reply:
ok I saw they're not declared, just a nullptr